cluster-operator/src/main/java/io/strimzi/operator/cluster/model/ZookeeperCluster.java

```diff
@@ -104,6 +106,30 @@ public static String headlessServiceName(String cluster) {
         return cluster + ZookeeperCluster.HEADLESS_SERVICE_NAME_SUFFIX;
     }
 
+    private static final Pattern POD_NAME_PATTERN = Pattern.compile("^(.*)-zookeeper-([0-9]+)$");
+
+    private static String podDnsName(String namespace, String cluster, String podName) {
+        return String.format("%s.%s.%s.svc.%s",
+                podName,
+                ZookeeperCluster.headlessServiceName(cluster),
+                namespace,
+                ModelUtils.KUBERNETES_SERVICE_DNS_DOMAIN);
+    }
+
+    public static String podDnsName(String namespace, String cluster, int podId) {
+        return podDnsName(namespace, cluster, ZookeeperCluster.zookeeperPodName(cluster, podId));
+    }
+
+    public static String podDnsName(String namespace, String podName) {
```

Comment:
Why does this seem so much more sophsticated then the `podDnsName` in `KafkaCluster`?

Reply:
Right now we only need the Kafka `podDnsName` in one place (where we generate the certificate for the broker). But for ZK we need it both in cert generation and when finding the leader, and we don't have the same info to hand as both call sites. I realise now that I can simplify this though. We don't have to extract the cluster name from the pod name, since we have it in the pod's `strimzi.io/cluster` label, so it should be possible to simplify the ZK case to only need a single method, as we're doing with Kafka.